### PR TITLE
Make models/taskinstance.py pylint compatible

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -1135,7 +1135,11 @@ def _run_raw_task(
             session.merge(self)
         session.commit()
 
-    def _prepare_and_execute_task_with_callbacks(self, context, session, task):
+    def _prepare_and_execute_task_with_callbacks(  # pylint: disable=unused-argument
+            self,
+            context,
+            session,
+            task):
```
Comment (turbaszek) [inline airflow/models/taskinstance.py]: Should we remove the argument then? What is the reason to keep it?

**Comment by turbaszek** [comment]
Is this another attempt to fix cyclic imports? https://github.com/apache/airflow/pull/9674

File: airflow/models/taskinstance.py [hunk 2]
```diff
@@ -442,16 +442,16 @@ def generate_command(dag_id: str,     # pylint: disable=too-many-arguments
     def log_filepath(self):
         """Filepath for TaskInstance"""
         iso = self.execution_date.isoformat()
-        log = os.path.expanduser(conf.get('logging', 'BASE_LOG_FOLDER'))
+        the_log = os.path.expanduser(conf.get('logging', 'BASE_LOG_FOLDER'))
         return ("{log}/{dag_id}/{task_id}/{iso}.log".format(
-            log=log, dag_id=self.dag_id, task_id=self.task_id, iso=iso))
+            log=the_log, dag_id=self.dag_id, task_id=self.task_id, iso=iso))
```
Comment (turbaszek) [inline airflow/models/taskinstance.py]: Should we use f-string?

File: airflow/models/taskinstance.py [hunk 3]
```diff
@@ -1762,19 +1767,18 @@ def filter_for_tis(
...
+                                    TaskInstance.execution_date == tik.execution_date)
                                for tik in tis])
             return or_(*filter_for_tis)
         if all(isinstance(t, TaskInstance) for t in tis):
-            filter_for_tis = ([and_(TI.dag_id == ti.dag_id,
-                                    TI.task_id == ti.task_id,
-                                    TI.execution_date == ti.execution_date)
+            filter_for_tis = ([and_(TaskInstance.dag_id == ti.dag_id,
+                                    TaskInstance.task_id == ti.task_id,
+                                    TaskInstance.execution_date == ti.execution_date)
```
Comment (turbaszek) [inline airflow/models/taskinstance.py:1770]: If we decide on long names then we should remove TR from global scope, shouldn't we?

Comment (potiuk) [reply airflow/models/taskinstance.py:1770]: > Is this another attempt to fix cyclic imports? #9674

It was not supposed to be, but it turned out to be exactly this :).

Comment (potiuk) [reply airflow/models/taskinstance.py:1770]: Yes.

File: airflow/models/taskinstance.py (hunk 2 above)
Comment (potiuk) [reply airflow/models/taskinstance.py]: Why not :)

**Comment by potiuk** [comment]
This one is still draft. Turned out that there is another cyclic dependency hidden in this one :).

File: airflow/models/taskinstance.py (hunk 3 above)
Comment (potiuk) [reply airflow/models/taskinstance.py:1770]: Ahh. I fixed it in a simple way I think :)

File: airflow/models/taskinstance.py [hunk 4]
```diff
@@ -1135,23 +1134,16 @@ def _run_raw_task(
...
-        from airflow.models.renderedtifields import RenderedTaskInstanceFields as RTIF
-        from airflow.sensors.base_sensor_operator import BaseSensorOperator
+        from airflow.models.renderedtifields import RenderedTaskInstanceFields
 
         task_copy = task.prepare_for_execution()
-        # Sensors in `poke` mode can block execution of DAGs when running
-        # with single process executor, thus we change the mode to`reschedule`
-        # to allow parallel task being scheduled and executed
-        if (
-            isinstance(task_copy, BaseSensorOperator) and
```
Comment (potiuk) [inline airflow/models/taskinstance.py:1150]: Hey @turbaszek -> I think that was the root cause for the cyclic import but it should be hopefully fixed now (at least SkipMixin -> TaskInstance -> BaseSensorOperator -> SkipMixin).

The solution was overloading prepare_for_execution method in BaseSensorOperator (seems also the right thing to do) but I would love you to take a look as well.

Comment (potiuk) [reply airflow/models/taskinstance.py:1150]: That is an interesting one :)

File: airflow/sensors/base_sensor_operator.py [hunk 5]
```diff
@@ -159,6 +160,16 @@ def _get_next_poke_interval(self, started_at, try_number):
         else:
             return self.poke_interval
 
+    def prepare_for_execution(self) -> BaseOperator:
```
Comment (turbaszek) [inline airflow/sensors/base_sensor_operator.py:163]:
```suggestion
    def prepare_for_execution(self) -> "BaseSensorOperator":
```
Lovely! Now we have this method it's indeed easier to add some custom "pre_execution" logic.

File: airflow/sensors/base_sensor_operator.py [hunk 6]
```diff
@@ -159,6 +160,16 @@ def _get_next_poke_interval(self, started_at, try_number):
...
+        task = super().prepare_for_execution()
+        # Sensors in `poke` mode can block execution of DAGs when running
+        # with single process executor, thus we change the mode to`reschedule`
+        # to allow parallel task being scheduled and executed
+        if conf.get('core', 'executor') == "DebugExecutor":
```
Comment (turbaszek) [inline airflow/sensors/base_sensor_operator.py:168]:
```suggestion
        if conf.get('core', 'executor') == "DebugExecutor":
```
Should we use `ExecutorLoader.DEBUG_EXECUTOR`?

**Review by turbaszek — APPROVED** [review]
This rocks 🎸

File: airflow/sensors/base_sensor_operator.py (hunk 5 above)
Comment (potiuk) [reply airflow/sensors/base_sensor_operator.py:163]: Incompatible return value :(

File: airflow/sensors/base_sensor_operator.py (hunk 6 above)
Comment (potiuk) [reply airflow/sensors/base_sensor_operator.py:168]: I have a problem with importing it as constants (MyPy complains). I prefer to leave it :(

**Pull request merged**